### ISSUE TITLE
[IntelliJ][ReportedByUser] Uncaught Exception  java.lang.NullPointerException

### ISSUE THREAD

**Issue by v-xiaocw** [microsoft/azure-tools-for-java#2414]
Repro steps:
1.Create a new project 
2.Sign in/sign out

IntelliJ build version: 2018.3.1 IC-183.4588.27
OS: Windows 10
JDK: JetBrains s.r.o 1.8.0_152-release
Plugin version: 3.15.0-183.4284.118
Additional Info: None
Parent component: 
```
com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog$DialogRootPane[,8,31,824x494,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=449,maximumSize=,minimumSize=,preferredSize=]
```
Error message: 
```
IdeaLoggingEvent[message=null, throwable=java.lang.NullPointerException
	at com.intellij.openapi.wm.impl.InternalDecorator.setTitleActions(InternalDecorator.java:316)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.setTitleActions(ToolWindowImpl.java:309)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.lambda$addToolbarItems$2(ServerExplorerToolWindowFactory.java:370)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.notifySignOutEventListener(AuthMethodManager.java:106)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.signOut(AuthMethodManager.java:159)
	at com.microsoft.azuretools.ijidea.actions.AzureSignInAction.onAzureSignIn(AzureSignInAction.java:114)
	at com.microsoft.intellij.serviceexplorer.azure.SignInOutAction$1.actionPerformed(SignInOutAction.java:46)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener.actionPerformedAsync(NodeActionListener.java:72)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeAction.fireNodeActionEvent(NodeAction.java:60)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.lambda$createPopupMenuForNode$0(ServerExplorerToolWindowFactory.java:211)
	at javax.swing.AbstractBu

**Comment by v-xiaocw** [comment]
java.lang.NullPointerException
	at com.intellij.openapi.wm.impl.InternalDecorator.setTitleActions(InternalDecorator.java:316)
	at com.intellij.openapi.wm.impl.ToolWindowImpl.setTitleActions(ToolWindowImpl.java:309)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.lambda$addToolbarItems$2(ServerExplorerToolWindowFactory.java:370)
	at com.microsoft.azuretools.authmanage.AuthMethodManager.notifySignInEventListener(AuthMethodManager.java:97)
	at com.microsoft.azuretools.ijidea.actions.AzureSignInAction.doSignIn(AzureSignInAction.java:133)
	at com.microsoft.azuretools.ijidea.actions.AzureSignInAction.onAzureSignIn(AzureSignInAction.java:117)
	at com.microsoft.intellij.serviceexplorer.azure.SignInOutAction$1.actionPerformed(SignInOutAction.java:46)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener.actionPerformedAsync(NodeActionListener.java:72)
	at com.microsoft.tooling.msservices.serviceexplorer.NodeAction.fireNodeActionEvent(NodeAction.java:60)
	at com.microsoft.intellij.components.ServerExplorerToolWindowFactory.lambda$createPopupMenuForNode$0(ServerExplorerToolWindowFactory.java:211)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)
	at java.awt.Component.processMouseEvent(Component.java:6548)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6313)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4903)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4526)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4725)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:719)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:664)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:363)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

**Comment by Flanker32** [comment]
We didn't remove the tool window listener when close a project, fixed in #3431
